src/components/Calculator.js

```diff
@@ -1,76 +1,180 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './style.css';
+import calculate from '../logic/calculator';
 
-export default class CalcApp extends React.Component {
+export default class CalcApp extends Component {
   constructor(props) {
     super(props);
-
-    this.state = {
-      res: 0,
-    };
+    this.state = {};
+    this.handleClick = this.handleClick.bind(this);
   }
 
+  handleClick = (buttonName) => {
+    this.setState((state) => calculate(state, buttonName));
+  };
+
+  showSolution = () => {
+    const { total, next, operation } = this.state;
+    const show = (total || '') + (operation || '') + (next || '');
+
+    return (show || '0');
+  };
+
   render() {
-    const { res } = this.state;
     return (
       <div className="wrapper">
-        <span className="operation-result">{res}</span>
+        <span className="operation-result">{this.showSolution()}</span>
         <section className="main-section">
-          <button type="button" className="left-most cell1">
+          <button
+            name="AC"
+            type="button"
+            className="left-most cell1"
+            onClick={(ev) => this.handleClick(ev.target.name)}
+          >
             AC
           </button>
-          <button type="button" className="left-most cell2">
+          <button
+            name="+/-"
+            type="button"
+            className="left-most cell2"
+            onClick={(ev) => this.handleClick(ev.target.name)}
+          >
             +/-
           </button>
-          <button type="button" className="left-most cell3">
+          <button
+            name="%"
+            type="button"
+            className="left-most cell3"
+            onClick={(ev) => this.handleClick(ev.target.name)}
+          >
             %
           </button>
-          <button type="button" className="right-most cell4">
+          <button
+            name="/"
+            type="button"
+            className="right-most cell4"
+            onClick={(ev) => this.handleClick(ev.target.name)}
+          >
             /
```

Comment:
- [ ] Please work on the functionality of this event, currently when I try to use `/` I get this error
![div](https://user-images.githubusercontent.com/44745855/129928374-d725e8e1-631e-4e09-acb9-e41323d32f09.png)

N.B: Remember that when you divide a number by `zero` let us say `5/0`
you are supposed to get `error` in the screen

Reply:
Thank you for the review. I have made the necessary changes

Reply:
- This issue is well addressed! :+1: :clap: